### PR TITLE
Pip functionality

### LINKED ISSUE #693: Picture in Picture Mode not working
**Describe the bug**
Picture in Picture mode is not working in CB online App

**To Reproduce**
1. Go to any course
2. Start any course
3. Click home button, it should start in picture in picture mode but it is not working

**Expected behavior**
Click home button, it should start in picture in picture mode but it is not working

Device :         Samsung J7 Prime
Android Version : Oreo 8.1
**Would you like to work on the issue?**

- [x] Yes
- [ ] No
- Other:

### PR DESCRIPTION
**Pull request by prathamk22**
Fixes #693 

Changes: 
Added Picture in Picture mode functionality.
Added play and pause button for better interaction in PIP Mode.
The video is not showing in picture but in device it is working perfectly. 

Screenshots for the change:
vlcsnap-2020-05-14-21h30m23s739

### CODE REVIEW

**Comment by aggarwalpulkit596** [comment]
Can you rebase your PR first ?

**Comment by prathamk22** [comment]
> Can you rebase your PR first ?

Rebased sir

**Comment by prathamk22** [comment]
@aggarwalpulkit596 please review it and let me know if it is approved or not.

**Comment by aggarwalpulkit596** [comment]
This one is relatively bigger PR I'll check it out locally till then you can work on other issues

**Comment by prathamk22** [comment]
> This one is relatively bigger PR I'll check it out locally till then you can work on other issues

Sure

**Comment by aggarwalpulkit596** [comment]
@prathamk22 you've added some font files please remove them

**Comment by aggarwalpulkit596** [comment]
@prathamk22 also if I press back button I do not go back to the previous activity instead app exits so you need fix that too

**Comment by aggarwalpulkit596** [comment]
And in case of a youtube video, the system hangs so I guess you need to take a look at this one thoroughly.Ping me when you're done

**Comment by prathamk22** [comment]
@aggarwalpulkit596 pushed with changes and in case of youtube video, it is working perfectly now.

**Review by aggarwalpulkit596 — DISMISSED** [review]
LGTM

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -164,8 +164,14 @@
             android:parentActivityName=".dashboard.DashboardActivity" />
         <activity
             android:name=".mycourse.player.VideoPlayerActivity"
+            android:taskAffinity=".mycourse.player.VideoPlayerActivity"
+            android:parentActivityName=".mycourse.MyCourseActivity"
+            android:autoRemoveFromRecents="true"
```
Comment (aggarwalpulkit596) [inline app/src/main/AndroidManifest.xml]: Why have you added both of them i.e exclude and autoRemove ?

Comment (prathamk22) [reply app/src/main/AndroidManifest.xml]: Sir because we don't want to show it in recent tasks list

**Comment by aggarwalpulkit596** [comment]
@prathamk22 I'm a bit dicy about the implementation will take another look.

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (prathamk22) [reply app/src/main/AndroidManifest.xml]: Sir I got the ans in https://medium.com/mindorks/android-picture-in-picture-mode-and-backstack-management-for-multiple-activities-8427fe3f8102 about that backpressed problem

Comment (aggarwalpulkit596) [reply app/src/main/AndroidManifest.xml]: https://stackoverflow.com/questions/44363530/difference-between-autoremovefromrecents-and-excludefromrecents
first of all, not both of them will be required and for instance and secondly take a look at youtube or google maps implementation of PIP

Comment (prathamk22) [reply app/src/main/AndroidManifest.xml]: So should I keep only one?

**Comment by aggarwalpulkit596** [comment]
Yeah i mean one of them would do the job

**Comment by aggarwalpulkit596** [comment]
@prathamk22 also one major issue is that when pip is not enabled and If i minimize the app for some reason I cannot go back so you need to figure out something else for this

**Comment by prathamk22** [comment]
> @prathamk22 also one major issue is that when pip is not enabled and If i minimize the app for some reason I cannot go back so you need to figure out something else for this

@aggarwalpulkit596 yes sir the thing is that VideoPlayerActivity is not going in the recents in any way, so when you minimise, it takes back you to the course fragment. I searched about it and found that many people implemented same and also said that we can't have them back when we press the home button and PIP is not activated. Then thing is when we go into PIP mode, it starts a new task and when we go back to the normal screen and press home button, as being a single task it ends and when we exclude it from recents it just take you to the last task. I researched about it alot and came up with this solution.

**Comment by aggarwalpulkit596** [comment]
@prathamk22 can you do one thing that if pip is disabled video player activity should go in recent otherwise not

**Comment by prathamk22** [comment]
@aggarwalpulkit596 changes done as per feedback

File: app/src/main/java/com/codingblocks/cbonlineapp/mycourse/content/CourseContentFragment.kt [hunk 2]
```diff
@@ -330,12 +332,18 @@ class CourseContentFragment : BaseCBFragment(), AnkoLogger, DownloadStarter {
                         LECTURE ->
                             if (contentLecture.lectureUid.isNotEmpty())
                                 startActivity(
-                                    intentFor<VideoPlayerActivity>(
-                                        CONTENT_ID to ccid,
-                                        SECTION_ID to sectionId
-                                    )
-                                )
-                            else
+                                    if (getPrefs()?.SP_PIP!!) {
```
Comment (aggarwalpulkit596) [inline app/src/main/java/com/codingblocks/cbonlineapp/mycourse/content/CourseContentFragment.kt]: can you use prefs from ViewModel instead just to make the consistency throughout the code

**Comment by aggarwalpulkit596** [comment]
WhatsApp Image 2020-05-29 at 1 29 31 PM
when the pip is disabled this is happening and it shouldn't be like this there should be only one activity no matter what
 also because of that app exist on the back press.

**Comment by aggarwalpulkit596** [comment]
Also, I would suggest making a method to open the activity like this 
https://github.com/coding-blocks/CBOnlineApp/blob/76dedab5445ec48fb0c9380338a808f0b3233476/app/src/main/java/com/codingblocks/cbonlineapp/dashboard/DashboardActivity.kt#L389
otherwise, we have to handle this everywhere manually

File: app/src/main/java/com/codingblocks/cbonlineapp/mycourse/content/CourseContentFragment.kt (hunk 2 above)
Comment (prathamk22) [reply app/src/main/java/com/codingblocks/cbonlineapp/mycourse/content/CourseContentFragment.kt]: Sir everywhere it is used like getPrefs(requireContext()). So I used like this only

**Comment by prathamk22** [comment]
@aggarwalpulkit596 sir changes done as per feedback. It is showing one activity in recent tasks.

File: app/src/main/java/com/codingblocks/cbonlineapp/mycourse/player/VideoPlayerActivity.kt [hunk 3]
```diff
@@ -786,5 +887,11 @@ class VideoPlayerActivity : BaseCBActivity(), EditNoteClickListener, AnkoLogger,
                 VIDEO_POSITION to position,
                 SECTION_ID to sectionId).singleTop()
         }
+
+        fun createVideoPlayerActivity(context: Context, contentId: String, sectionId: String, exclude: Boolean): Intent {
```
Comment (aggarwalpulkit596) [inline app/src/main/java/com/codingblocks/cbonlineapp/mycourse/player/VideoPlayerActivity.kt]: also you can use the exisiting method in that case 
https://github.com/coding-blocks/CBOnlineApp/blob/688f7fdfe99e5cce1f4f2dd120a1552686a62ed8/app/src/main/java/com/codingblocks/cbonlineapp/mycourse/player/VideoPlayerActivity.kt#L884

**Review by aggarwalpulkit596 — CHANGES_REQUESTED** [review]

**Review by aggarwalpulkit596 — APPROVED** [review]

**Pull request merged**